lib/cache/add-remote-git.js

```diff
@@ -173,97 +176,124 @@ function fetchRemote (p, u, co, origUrl, cb) {
   )
 }
 
-function resolveHead (p, u, co, origUrl, cb) {
+function resolveHead (params, cb) {
+  log.verbose('resolveHead', 'original treeish', params.treeish)
   git.whichAndExec(
-    [ "rev-list", "-n1", co ],
-    { cwd : p, env : gitEnv() },
+    ['rev-list', '-n1', params.treeish],
+    { cwd: params.checkoutPath, env: gitEnv() },
     function (er, stdout, stderr) {
-      stdout = (stdout + "\n" + stderr).trim()
+      params.exactTreeish = (stdout + '\n' + stderr).trim()
```

Comment:
Mutating params feels icky. And again, I'm left wondering why stderr is glommed on here?